jpo-geojsonconverter/src/main/java/us/dot/its/jpo/geojsonconverter/converter/tim/TimProcessedJsonConverter.java

```diff
@@ -0,0 +1,1777 @@
+                .filter(text -> text != null && !text.trim().isEmpty()).collect(Collectors.joining(" "));
+        timContent.setSentence(combinedMessage);
+
+        return timContent;
+    }
+
+    /**
+     * Process advisory content.
+     *
+     * @param advisoryList List of advisory sequences
+     * @param contentItems List to store ordered content items
+     */
+    private void processAdvisoryContent(List<ITIScodesAndTextSequence> advisoryList,
+            List<ProcessedTimContentItem> contentItems) {
+        for (ITIScodesAndTextSequence advisory : advisoryList) {
+            processItisAndTextSequence(advisory.getItem(), contentItems);
+        }
+    }
+
+    /**
+     * Process speed limit content.
+     *
+     * @param speedLimitList List of speed limit sequences
+     * @param contentItems List to store ordered content items
+     */
+    private void processSpeedLimitContent(List<SpeedLimitSequence> speedLimitList,
+            List<ProcessedTimContentItem> contentItems) {
+        for (SpeedLimitSequence speedLimit : speedLimitList) {
+            processItisAndTextSequence(speedLimit.getItem(), contentItems);
+        }
+    }
+
+    /**
+     * Process work zone content.
+     *
+     * @param workZoneList List of work zone sequences
+     * @param contentItems List to store ordered content items
+     */
+    private void processWorkZoneContent(List<WorkZoneSequence> workZoneList,
+            List<ProcessedTimContentItem> contentItems) {
+        for (WorkZoneSequence workZone : workZoneList) {
+            processItisAndTextSequence(workZone.getItem(), contentItems);
+        }
+    }
+
+    /**
+     * Process ITIS and text sequence from various content types.
+     *
+     * @param item The item choice containing ITIS and text data
+     * @param contentItems List to store ordered content items
+     */
+    private void processItisAndTextSequence(Object item, List<ProcessedTimContentItem> contentItems) {
+        if (item == null) {
+            return;
+        }
+
+        // Handle different item types using reflection-like approach
+        try {
+            if (item instanceof ITIScodesAndTextSequence.ItemChoice) {
+                ITIScodesAndTextSequence.ItemChoice itemChoice = (ITIScodesAndTextSequence.ItemChoice) item;
+                processItisAndTextItem(itemChoice.getItis(), itemChoice.getText(), contentItems);
+            } else if (item instanceof SpeedLimitSequence.ItemChoice) {
+                SpeedLimitSequence.ItemChoice itemChoice = (SpeedLimitSequence.ItemChoice) item;
+                processItisAndTextItem(itemChoice.getItis(), itemChoice.getText(), contentItems);
+            } else if (item instanceof WorkZoneSequence.ItemChoice) {
+                WorkZoneSequence.ItemChoice itemChoice = (WorkZoneSequence.ItemChoice) item;
+                processItisAndTextItem(itemChoice.getItis(), itemChoice.getText(), contentItems);
+            }
```

Comment:
suggestion: Use pattern matching syntax instead of casting to simplify this eg

```java
            if (item instanceof ITIScodesAndTextSequence.ItemChoice itemChoice) {
                processItisAndTextItem(itemChoice.getItis(), itemChoice.getText(), contentItems);
            } else if ...
```